app/filtering.js

```diff
@@ -495,9 +495,14 @@ function registerPermissionHandler (session, partition) {
         }
       }
 
+      let permissionUrl = origin
+      if (permission === 'openExternal') {
+        permissionUrl = mainFrameUrl
```

Comment:
i think you want to set this in https://github.com/brave/browser-laptop/pull/13554/files#diff-8a9bac13ef6264feb2b6da7c18d86b3cR451 so that `settings` and `tempSettings` are the correct settings for the origin